test/snippets.cpp

```diff
@@ -29,6 +29,73 @@ usingStrings()
 
         string str2( make_storage<pool>() ); // empty string, pool storage
 
+        //]
+    }
+    {
+        //[snippet_strings_2
+
+        std::string sstr1 = "helloworld";
+        std::string sstr2 = "world";
+
+        json::string jstr1 = "helloworld";
+        json::string jstr2 = "world";
+
+
+        jstr2.insert(0, jstr1.subview(0, 6));
+        // now contains "helloworld"
+        
+        // this is equivalent to
+
+        sstr1.insert(0, sstr2, 0, 6);
+        // now contains "helloworld"
+
+        //]
+    }
+    {
+        //[snippet_strings_3
+
+        std::string sstr = "hello";
+
+        json::string jstr = "hello";
+
+        sstr.append({'w', 'o', 'r', 'l', 'd'});
+        // now contains "helloworld"
+
+        // such syntax is inefficient, and the same can
+        // be achieved with a character array.
+
+        jstr.append("world");
+        // now contains "helloworld"
+
+        //]
+    }
+
+    {
+        //[snippet_strings_4
+
+        json::string str = "Boost.JSON";
+        json::string_view sv = str;
+
+        // all of these call compare(string_view)
+        str.compare(sv);
```

Comment:
make these `assert`s (not `BOOST_ASSERT`), e.g.:
```
assert( str.compare(sv) < 0 );
```

Reply:
For the other examples specifying the post-operation state of the string I added asserts. No asserts are needed here though.